tests/integration_tests/test_gym_envs.py

```diff
@@ -0,0 +1,130 @@
+    "HandManipulateBlockDense-v0", "HandManipulateEggRotateDense-v0",
+    "HandManipulateEggFullDense-v0", "HandManipulateEggDense-v0",
+    "HandManipulatePenRotateDense-v0", "HandManipulatePenFullDense-v0",
+    "HandManipulatePenDense-v0"
+]
+
+
+class TestGymEnvs(unittest.TestCase):
+
+    garage_envs = [TfEnv, TheanoEnv]
+
+    def _routine_test_pickling_wrapper(self, garage_env, spec):
+        env = garage_env(env_name=spec.id)
+
+        # Roundtrip serialization
+        env = pickle.loads(pickle.dumps(env))
+
+        # Step and render still work after pickling
+        env.reset()
+        env.step(env.action_space.sample())
+
+        try:
+            env.render()
+        except NotImplementedError:
+            assert spec.id in not_rendering_envs, "It seems environment %s " \
+                    "cannot be rendered. Add it to the corresponding white" \
+                    "list in test_gym_envs.py." % (spec.id)
+
+        # Test serialization once the viewer object has been set in env, but
+        # don't reassign env with the one produced with pickle to avoid
+        # losing the reference to the viewer object which is required in
+        # env.close()
+        pickled_env = pickle.loads(pickle.dumps(env))
+        assert pickled_env
+
+        try:
+            env.close()
+        except AttributeError:
+            assert spec.id in not_closing_envs, "It seems environment %s " \
+                    "cannot be closed. Add it to the corresponding white" \
+                    "list in test_gym_envs.py." % (spec.id)
+
+    @params(*list(product(garage_envs, list(gym.envs.registry.all()))))
+    def test_pickling_wrapper(self, garage_env, spec):
+        # Skip Atari environments
+        if spec._entry_point == "gym.envs.atari:AtariEnv":
+            return
+        process = Process(
+            target=self._routine_test_pickling_wrapper,
+            args=(
+                garage_env,
+                spec,
+            ))
+        process.start()
+        process.join()
+        assert not process.exitcode
+
+    def _routine_test_step_and_render(self, garage_env, spec):
+        env = garage_env(env_name=spec.id)
+
+        # Step and render work
+        env.reset()
+        done = False
+        for _ in range(10):
+            _, _, done, _ = env.step(env.action_space.sample())
+            try:
+                env.render()
+            except NotImplementedError:
+                assert spec.id in not_rendering_envs, "It seems environment " \
+                        "%s cannot be rendered. Add it to the corresponding " \
+                        "white list in test_gym_envs.py." % (spec.id)
+            if done:
+                break
+
+        try:
+            env.close()
+        except AttributeError:
+            assert spec.id in not_closing_envs, "It seems environment %s " \
+                    "cannot be closed. Add it to the corresponding white" \
```

Comment:
Remove this. The future programmer should root cause it (and read the test) before reflexively whitelisting it.